### DISCUSSION TITLE
Additional Subnet apart from Public, Private App and Persistent

### DISCUSSION THREAD

**Discussion by vramanufidato** [gruntwork-io/knowledge-base#636] [How do I] [answered]
Hi Team,

Currently we can see 1) Public, 2) Private App and 3) Persistent Subnets [Supported Subnets](https://docs.gruntwork.io/guides/build-it-yourself/vpc/core-concepts/subnets)are been supported and the logic is present in the VPC modules. If we need to add additional subnet apart from these what is the best approach. 
As per documentation extend the catalog its mentioned but not sure if this is right approach for this requirement. [extend-service](https://docs.gruntwork.io/reference/services/intro/create-your-own-service-catalog#extend-gruntwork-services)

---

<ins datetime="2023-01-17T11:54:23Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109798">Tracked in ticket #109798</a></p>
</ins>

**Comment by Etiene** [comment, marked as answer]
Yes, extending by wrapping would be the correct approach. You might have to customize the cidr blocks, but you can use the outputs from the vpc-app module to create the necessary extra subnets & routes.
